patches.go

```diff
@@ -133,39 +84,23 @@ func ApplyFromFieldPathPatch(p PatchInterface, from, to runtime.Object) error {
 }
 
 // ApplyCombineFromVariablesPatch patches the "to" resource, taking a list of
-// input variables and combining them into a single output value.
-// The single output value may then be further transformed if they are defined
-// on the patch.
+// input variables and combining them into a single output value. The single
+// output value may then be further transformed if they are defined on the
+// patch.
 func ApplyCombineFromVariablesPatch(p PatchInterface, from, to runtime.Object) error {
-	// Combine patch requires configuration
-	if p.GetCombine() == nil {
-		return errors.Errorf(errFmtRequiredField, "Combine", p.GetType())
-	}
-	// Destination field path is required since we can't default to multiple
-	// fields.
-	if p.GetToFieldPath() == "" {
-		return errors.Errorf(errFmtRequiredField, "ToFieldPath", p.GetType())
-	}
-
-	combine := p.GetCombine()
-	vl := len(combine.Variables)
-
-	if vl < 1 {
-		return errors.New(errCombineRequiresVariables)
-	}
```

Comment:
Not strictly related to this PR, but I noticed this _wasn't_ already being handled by `ValidateResources`, but it should be. I updated `validate.go` to add more thorough validation of combine patches.